### ISSUE TITLE
System.NullReferenceException with Query Syntax error

### ISSUE THREAD

**Issue by dabbers** [ChilliCream/hotchocolate#466]
**Describe the bug**
When performing a query with invalid syntax:
"{hello(name:\"dab\"}"
I get a nullreference exception.
   at HotChocolate.Execution.DiagnosticMiddleware.InvokeAsync(IQueryContext context)
   at HotChocolate.Execution.QueryExecuter.ExecuteMiddlewareAsync(IQueryContext context)
   at dab.graphql.Program.doAsyncThing() in C:\Users\me\source\repos\dab.graphql\dab.graphql\Program.cs:line 32

**To Reproduce**
Following the getting started. I created a new console app. Nuget fetched the hotchocolate pre-release, create a QueryType/Query objects with a Hello(string name) method. Then I attempt to execute the above query. It throws an exception breaking the program.

**Expected behavior**
I expect to have a JSON error[] return instead complaining about an invalid syntax.

**Desktop (please complete the following information):**
 - OS:  Win10
 - Version 0.7.0-preview.26

**Comment by dabbers** [comment]
Code repro (pretty simple):

```
            var schema = Schema.Create
            (
                c =>
                {
                    c.RegisterType
                    (
                        new QueryType()
                    );
                }

            );

            Console.WriteLine(schema.ToString());
            var executer = schema.MakeExecutable();

            var t = await executer.ExecuteAsync
            (
                new HotChocolate.Execution.QueryRequest
                (
                    "{hello(name:\"dab\"}"
                ), 
                new System.Threading.CancellationToken()
            );
            Console.WriteLine(t.ToString());
```

**Comment by michaelstaib** [comment]
Actually this should throw a schema exception tha explains that objets must have a name. I will look intobit and issue a fix. Thanks for reporting this one.

**Comment by michaelstaib** [comment]
Sorry, I misread the code... could you provide the query type?

**Comment by michaelstaib** [comment]
I misread your issue.... you are absolutely right... there should be a syntax exception in the response... let me check why this happens.

**Comment by dabbers** [comment]
Last night, I created a quick integration test and have been stepping through the code to understand the issue and I think I understand the problem:
During the `ParseQueryMiddleware` we call the default parser and parse the document. We hit a syntax error, and throw a `syntaxexception`. This Exception pops out of the middleware callstack until we hit the `ExceptionMiddleware`. This catches the exception, stores it in the errors, then continues out of the call stack until we land back in Diagnostic middleware where we expect `context.ValidationResult` to be populated already. I played with some solutions for a bit but I realize I don't yet understand how your error reporting works quite yet (Is `QueryError` the correct exception for a syntax error? Should a new exception type be added for syntax error? Does the Code `ErrorProperty` need to be filled in? I was playing with this as a fix in `ExceptionMiddleware`,

```
            catch(Language.SyntaxException ex)
            {
                context.Exception = ex;
                context.Result = QueryResult.CreateError(
                    _errorHandler.Handle(
                        new QueryError(
                            ex.Message,
                            new Location[] { new Location(ex.Line, ex.Column) },
                            new ErrorProperty("code", context.Request.Query))));
            }
```

With a change in `DiagnosticMiddleware` to include a:
```
                // If ValidationResult is null, we likely have an exception from the Parser
                if (context.ValidationResult != null && context.ValidationResult.HasErrors)
```

But I don't know if this is a good way to go about fixing this yet. I'll keep looking in the code to understand it more. I only have a basic understanding of the code so far, so hopefully looking at it longer will help.

Thanks!

**Comment by michaelstaib** [comment]
We have changed the `DiagnosticsMiddleware´ and the `NullReferenceException` should no more happen. But we still have to ensure that the syntax exception is correctly written to the result. Moreover, we need to add a diagnostics event for the parser.

**Comment by michaelstaib** [comment]
This one is now fixed with 0.7.0-preview.28
